ext/sysvshm/sysvshm.c

```diff
@@ -29,8 +29,44 @@
 #include "ext/standard/info.h"
 #include "ext/standard/php_var.h"
 #include "zend_smart_str.h"
+#include "Zend/zend_interfaces.h"
 #include "php_ini.h"
 
+/* SysvSharedMemory class */
+
+zend_class_entry *sysvshm_ce;
+static zend_object_handlers sysvshm_object_handlers;
+
+static inline sysvshm_shm *sysvshm_from_obj(zend_object *obj) {
+	return (sysvshm_shm *)((char *)(obj) - XtOffsetOf(sysvshm_shm, std));
+}
+
+#define Z_SYSVSHM_P(zv) sysvshm_from_obj(Z_OBJ_P(zv))
+
+static zend_object *sysvshm_create_object(zend_class_entry *class_type) {
+	sysvshm_shm *intern = zend_object_alloc(sizeof(sysvshm_shm), class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+	intern->std.handlers = &sysvshm_object_handlers;
+
+	return &intern->std;
+}
+
+static zend_function *sysvshm_get_constructor(zend_object *object) {
+	zend_throw_error(NULL, "Cannot directly construct SysvSharedMemory, use shm_attach() instead");
+	return NULL;
+}
+
+static void sysvshm_free_obj(zend_object *object)
+{
+	sysvshm_shm *sysvshm = sysvshm_from_obj(object);
+
+	shmdt((void *) sysvshm->ptr);
```

Comment:
I'd put this behind `if (sysvshm->ptr)`. I guess it works anyway, but the man page doesn't spec NULL behavior.